### PR TITLE
latest SocialIntegration

### PR DESCRIPTION
**Pull request by danacebryantalfonso**
Integrated Realm database and kwartracker signup and login UI

### CODE REVIEW

File: SocialIntegration/app/build.gradle [hunk 1]
```diff
@@ -25,16 +26,26 @@ android {
...
+}
+
 dependencies {
+
+    implementation 'androidx.navigation:navigation-fragment:2.3.4'
+    implementation 'androidx.navigation:navigation-ui:2.3.4'
+    implementation 'androidx.lifecycle:lifecycle-livedata-ktx:2.3.1'
     def navigation_fragment_ktx_version = "2.3.4"
     def navigation_ui_ktx_version = "2.3.4"
     def drawer_layout_version = "1.1.1"
```
Comment (rexendy-lig) [inline SocialIntegration/app/build.gradle:51]: pls create variable for the versions

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            bottomSheetBehavior.state = BottomSheetBehavior.STATE_HIDDEN
+            true
+        }
+
+        bottomSheetBehavior.state = BottomSheetBehavior.STATE_EXPANDED
+        /*fab.setOnClickListener {
+        val img: ImageView = findViewById(R.id.imgLogo)
+            img.visibility = View.GONE
+            bottomSheetBehavior.state = BottomSheetBehavior.STATE_EXPANDED
+        }*/
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt:88]: remove commented codes if not used

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_social_integration)
+        Realm.init(this)
+        val toolbar: Toolbar = findViewById(R.id.toolbar)
+        val drawerLayout: DrawerLayout = findViewById(R.id.drawer_layout)
+        val navView: NavigationView = findViewById(R.id.nav_view)
+        val navController = findNavController(R.id.nav_host_fragment)
+        val navigationView: NavigationView = findViewById(R.id.navigationView)
+        val bottomSheetBehavior = BottomSheetBehavior.from(navigationView)
+        val scrim: FrameLayout = findViewById(R.id.scrim)
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt:56]: pls use binding for binding the xml to activitiy

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+                    ) { `object`, _ ->
+                        launchSignIn()
+                        println("${`object`.toString()}")
+                    }
+                    val parameters = Bundle()
+                    parameters.putString("fields", "id,name,link")
+                    request.parameters = parameters
+                    request.executeAsync()
+
+
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt:120]: remove space

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+                }
+
+                override fun onCancel() {}
+                override fun onError(exception: FacebookException) {}
+
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt:125]: remove space

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+                s: CharSequence?, start: Int,
+                count: Int, after: Int
+            ) {
+            }
+
+            override fun onTextChanged(
+                s: CharSequence, start: Int,
+                before: Int, count: Int
+            ) {
+
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt:145]: remove space

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signin/SignInViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+class SignInViewModel : ViewModel() {
+
+    private val _text = MutableLiveData<String>().apply {
+        value = "This is signin Fragment"
+    }
+    val text: LiveData<String> = _text
+
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signin/SignInViewModel.kt:13]: remove space

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpFragment.kt [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        savedInstanceState: Bundle?
+    ): View? {
+        signUpViewModel =
+                ViewModelProvider(this).get(SignUpViewModel::class.java)
+        val root = inflater.inflate(R.layout.fragment_signup, container, false)
+        /*val textView: TextView = root.findViewById(R.id.text_slideshow)
+        signUpViewModel.text.observe(viewLifecycleOwner, Observer {
+            textView.text = it
+        })*/
+
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpFragment.kt:33]: remove commented code if not needed

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpFragment.kt [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val tvSignin: TextView = view.findViewById(R.id.tv_signin)
+        val txtUsername: EditText = view.findViewById(R.id.txt_username)
+        val txtPassword: EditText = view.findViewById(R.id.txt_password)
+        // Delete all persons
+        // Using executeTransaction with a lambda reduces code size and makes it impossible
+        // to forget to commit the transaction.
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpFragment.kt:44]: use /**/ for multiline comments

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpViewModel.kt [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+import androidx.lifecycle.ViewModel
+
+class SignUpViewModel : ViewModel() {
+
+    private val _text = MutableLiveData<String>().apply {
+        value = "This is signup Fragment"
+    }
+
+    val text: LiveData<String> = _text
+
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpViewModel.kt:14]: remove space

File: SocialIntegration/app/src/main/res/layout/content_main.xml [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        android:layout_height="match_parent"
+        android:background="@color/teal">
+        <FrameLayout
+            android:id="@+id/scrim"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent" >
+        </FrameLayout>
+
+        <com.google.android.material.navigation.NavigationView
+            android:id="@+id/navigationView"
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/res/layout/content_main.xml:20]: navigation_view

File: SocialIntegration/app/src/main/res/layout/fragment_signin.xml [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="vertical">
+
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:layout_width="match_parent"
+                android:layout_height="697dp">
+
+                <TextView
+                    android:id="@+id/textView2"
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/res/layout/fragment_signin.xml:21]: pls use proper naming of variable

File: SocialIntegration/app/src/main/res/layout/fragment_signin.xml [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="Welcome\nBack"
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/res/layout/fragment_signin.xml:24]: pls put all constant text to string.xml

File: SocialIntegration/app/src/main/res/layout/fragment_signin.xml [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+                    app:backgroundTint="#33bed6"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintHorizontal_bias="0.505"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toBottomOf="@+id/txt_password"
+                    app:layout_constraintVertical_bias="0.055" />
+
+                <TextView
+                    android:id="@+id/textView"
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/res/layout/fragment_signin.xml:109]: pls use proper naming of variable

File: SocialIntegration/app/src/main/res/layout/fragment_signin.xml [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+                    android:layout_marginBottom="164dp"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintHorizontal_bias="0.494"
+                    app:layout_constraintStart_toStartOf="parent">
+
+                </com.google.android.gms.common.SignInButton>
+
+                <TextView
+                    android:id="@+id/textView6"
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/res/layout/fragment_signin.xml:148]: pls use proper naming of variable

File: SocialIntegration/app/src/main/res/layout/fragment_signup.xml [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+    android:layout_height="match_parent"
+    tools:context=".ui.signup.SignUpFragment"
+    android:orientation="vertical">
+
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:layout_width="match_parent"
+                android:layout_height="697dp">
+
+                <TextView
+                    android:id="@+id/textView2"
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/res/layout/fragment_signup.xml:16]: pls use proper naming of varibale

File: SocialIntegration/app/src/main/res/layout/fragment_signup.xml [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+                    android:layout_width="match_parent"
+                    android:layout_height="60dp"
+                    android:background="@drawable/border_top"
+                    android:gravity="center"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintStart_toStartOf="parent">
+
+                    <TextView
+                        android:id="@+id/textView8"
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/res/layout/fragment_signup.xml:162]: pls use proper naming of variable

**Review by rexendy-lig — CHANGES_REQUESTED** [review]

File: SocialIntegration/app/build.gradle (hunk 1 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/build.gradle:51]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt (hunk 2 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt (hunk 3 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt (hunk 4 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt (hunk 5 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt (hunk 6 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/exercise/socialintegration/SocialIntegrationActivity.kt]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signin/SignInViewModel.kt (hunk 7 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signin/SignInViewModel.kt]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpFragment.kt (hunk 8 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpFragment.kt]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpFragment.kt (hunk 9 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpFragment.kt]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpViewModel.kt (hunk 10 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/exercise/socialintegration/ui/signup/SignUpViewModel.kt]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/res/layout/fragment_signin.xml (hunk 12 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/res/layout/fragment_signin.xml]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/res/layout/fragment_signin.xml (hunk 13 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/res/layout/fragment_signin.xml:23]: done in f5a921070510161d82817a5f1b4a5a002397e7c2

File: SocialIntegration/app/src/main/res/layout/fragment_signin.xml (hunk 15 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/res/layout/fragment_signin.xml]: done in ef7cf5c7963f284f8b7ece3e94ff0c230e015b03

File: SocialIntegration/app/src/main/res/layout/fragment_signup.xml (hunk 16 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/res/layout/fragment_signup.xml:16]: done in ef7cf5c7963f284f8b7ece3e94ff0c230e015b03

File: SocialIntegration/app/src/main/res/layout/fragment_signup.xml (hunk 17 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/res/layout/fragment_signup.xml]: done in ef7cf5c7963f284f8b7ece3e94ff0c230e015b03

**Review by rexendy-lig — APPROVED** [review]

**Pull request merged**